internal/gcsx/random_reader_test.go

```diff
@@ -938,7 +937,7 @@ func (t *RandomReaderTest) Test_ReadAt_CacheHitNextToCacheMissIncaseOfInvalidJob
 	_, cacheHit, err = t.rr.ReadAt(buf, 0)
 
 	ExpectEq(nil, err)
-	ExpectTrue(cacheHit)
+	ExpectFalse(cacheHit)
```

Comment:
@sethiay I am not sure if this should be a cache miss. Can you check once?

Reply:
i think it should be a cacheMiss because during rc2 read (2nd t.rr.ReadAt) call, the data is served from GCS and cache handle is closed as cache will give error because we have invalidated the cache before that. 
it is during the rc3 call that cache is populated again with new cache handle and because it is populated so should be a cache miss.

Reply:
Thanks for the explanation!